pennylane/pulse/convenience_functions.py

```diff
@@ -115,3 +116,134 @@ def f(p, t):
         return _f(p, t)
 
     return f
+
+
+def pwc(timespan):
+    """Creates a function that is piecewise-constant in time.
+
+    Args:
+            timespan(Union[float, tuple(float, float)]: The timespan defining the region where the function is non-zero.
+              If an integer is provided, the timespan is defined as ``(0, timespan)``.
+
+    Returns:
+            func: a function that takes two arguments, an array of trainable parameters and a `float` defining the
+            time at which the function is evaluated. When called, the function uses the array of parameters to
+            create evenly sized bins within the ``timespan``, with each bin value set by an element of the array.
+            It then selects the value of the parameter array corresponding to the specified time, based on the
+            assigned binning.
+
+    **Example**
+
+    >>> timespan = (1, 3)
+    >>> f1 = pwc(timespan)
+
+    The resulting function ``f1`` has the call signature ``f1(params, t)``. If passed an array of parameters and
+    a time, it will assign the array as the constants in the piecewise function, and select the constant corresponding
+    to the specified time, based on the time interval defined by ``timespan``.
+
+    >>> params = np.linspace(10, 20, 10)
```

Comment:
Could we use a list of values instead? I believe this will make it clear that the values of params are returned. For example:
```suggestion
    >>> params = [10, 11, 12, 13, 14]
```